pkg/scheduler/queue/queue.go

```diff
@@ -143,14 +143,14 @@ func (q *RequestQueue) dispatcherLoop() {
 			switch qe.operation {
 			case registerConnection:
 				q.connectedQuerierWorkers.Inc()
-				queues.addQuerierConnection(qe.querierID)
+				queues.tenantQuerierState.addQuerierConnection(qe.querierID)
```

Comment:
Should we perhaps keep the `addQuerierConnection`, `removeQuerierConnection` and `notifyQuerierShutdown` methods on the `queues` type, and have those pass the call through to `tenantQuerierState`? The fact that `queues` uses another type to manage this information seems like an implementation detail that `RequestQueue` doesn't need to know about.

Reply:
done